packages/web/src/shared/components/ConfirmationDialog.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  Fragment,
+  forwardRef,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import Button from './actions/Button';
+import TextInput from './form/TextInput';
+
+export interface ConfirmationDialogProps {
+  title: string;
+  description: string;
+  confirmationValue: string;
+}
+
+export interface ConfirmationDialogRef {
+  open(): Promise<boolean>;
+}
+
+const ConfirmationDialog = forwardRef<
+  ConfirmationDialogRef,
+  ConfirmationDialogProps
+>(({ title, description, confirmationValue }: ConfirmationDialogProps, ref) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const resolveOpen = useRef<
+    ((value: boolean | PromiseLike<boolean>) => void) | null
+  >(null);
+
+  useImperativeHandle(
+    ref,
+    () => ({
+      open: async () =>
+        new Promise<boolean>((resolve) => {
+          setIsOpen(true);
+          resolveOpen.current = resolve;
+        }),
+    }),
+    []
+  );
+
+  const { t } = useTranslation();
+  const [enableConfirmationButton, setEnableConfirmationButton] =
+    useState<boolean>(false);
+
+  function handleResult(result: boolean) {
+    setIsOpen(false);
+    if (resolveOpen.current) {
+      resolveOpen.current(result);
+    }
+  }
+
+  return (
+    <Transition appear show={isOpen} as={Fragment}>
+      <Dialog
+        as="div"
+        className="relative z-50"
+        onClose={() => handleResult(false)}
+      >
+        <Transition.Child
+          as={Fragment}
+          enter="ease-out duration-300"
+          enterFrom="opacity-0"
+          enterTo="opacity-100"
+          leave="ease-in duration-200"
+          leaveFrom="opacity-100"
+          leaveTo="opacity-0"
+        >
+          <div className="fixed inset-0 bg-black bg-opacity-25" />
+        </Transition.Child>
+
+        <div className="fixed inset-0 overflow-y-auto">
+          <div className="flex min-h-full items-center justify-center p-4 text-center">
+            <Transition.Child
+              as={Fragment}
+              enter="ease-out duration-300"
+              enterFrom="opacity-0 scale-95"
+              enterTo="opacity-100 scale-100"
+              leave="ease-in duration-200"
+              leaveFrom="opacity-100 scale-100"
+              leaveTo="opacity-0 scale-95"
+            >
+              <Dialog.Panel className="w-full max-w-md flex flex-col gap-2 transform overflow-hidden rounded-2xl bg-white p-6 text-left align-middle shadow-xl transition-all">
+                <Dialog.Title
+                  as="h3"
+                  className="text-lg font-medium leading-6 text-gray-900"
+                >
+                  {title}
+                </Dialog.Title>
+                <p className="text-sm text-red-500 font-bold">{description}</p>
+                <p id="prompt" className="text-sm text-gray-600">
+                  {t('confirm_prompt', { value: confirmationValue })}
+                </p>
+                <TextInput
```

Comment:
Can we wrap this in a Form, so we can get required validation and enter to submit? We'll also need to make the button a submit button

Reply:
Done